_includes/figure.html

```diff
@@ -1,23 +1,22 @@
-{%- assign img_path = include.path | remove: ".jpg" | remove: ".jpeg" | remove: ".png" | remove: ".tiff" -%}
+{%- assign img_path = include.path | remove: ".jpg" | remove: ".jpeg" | remove: ".png" | remove: ".tiff" | remove: ".gif" -%}
 
 <figure>
-
   <picture>
+    <!-- Auto scaling with imagemagick -->
+    <!-- See https://www.debugbear.com/blog/responsive-images#w-descriptors-and-the-sizes-attribute and 
+      https://developer.mozilla.org/en-US/docs/Learn/HTML/Multimedia_and_embedding/Responsive_images for info on defining 'sizes' for responsive images-->
     {% if site.imagemagick.enabled %}
-    {% for i in site.imagemagick.widths -%}
-      <source
-        class="responsive-img-srcset"
-        media="(max-width: {{ i }}px)"
-        srcset="{{ img_path | relative_url }}-{{ i }}.webp"
-      />
-    {% endfor -%}
+      <source class = "responsive-img-srcset" srcset="
+      {% for i in site.imagemagick.widths -%}
+      {{ img_path | relative_url }}-{{ i }}.webp {{i}}w,
+      {% endfor -%}" 
+      {% if include.sizes %} sizes="{{include.sizes}}" {% else %} sizes="95vw" {% endif %}
+      type="image/webp"/>
     {% endif %}
-
-    <!-- Fallback to the original file -->
     <img
       src="{% if include.cache_bust %}{{ include.path | relative_url | bust_file_cache }}{% else %}{{ include.path | relative_url }}{% endif %}"
       {% if include.class %}class="{{ include.class }}"{% endif %}
-      {% if include.width %}width="{{ include.width }}"{% else %}width="auto"{% endif %}
+      {% if include.width %}width="{{ include.width }}"{% else %}width="100%"{% endif %}
```

Comment:
Leaving width as auto led to weird scaling behavior, mainly visible on the example projects page, where the .img-fluid sets max-width of 100%, not width of 100%. Changing the default to 100% makes images predictably take up the full width available to them. Without it, the width of the image behaves oddly and is not predictable as it will be different depending which size image is loaded. See examples of odd widths below.

<img width="554" alt="image" src="https://github.com/alshedivat/al-folio/assets/68435033/55cc0dc7-65ba-40c4-a31e-2c0bca49fd2e">
<img width="463" alt="image" src="https://github.com/alshedivat/al-folio/assets/68435033/c8d73fc1-3db9-4540-8658-3016763bd03d">